src/bin/add/main.rs

```diff
@@ -70,20 +72,46 @@ crates.io registry suggests. One goal of `cargo add` is to prevent you from usin
 dependencies (version set to "*").
 "#;
 
+fn print_msg(dep: &Dependency, section: &[String], optional: bool) {
+    let optional = if optional { "optional " } else { "" };
+    let section = if section.len() == 1 {
+        section[0].clone()
+    } else {
+        format!("{} for target {}", &section[2], &section[1])
+    };
+
+    let mut output = StandardStream::stdout(ColorChoice::Auto);
+    output
+        .set_color(ColorSpec::new().set_fg(Some(Color::Green)).set_bold(true))
+        .unwrap();
```

Comment:
Use `?` instead of unwrap here (same for the reset).

When does this fail exactly? Imagine I do `cargo add foo >> add.log`. Will `set_color` do nothing then, because stdout is not a color terminal? (This is what I'd expect.)

Reply:
When stdout is not a terminal, it should do nothing (without an Error), since I've set the ColorChoice to "Auto".
It will only fail when the actual writing of the escape sequence to stdout fails (for example, when writing to a broken pipe that's also a terminal). Quite rare but I'll try to do error handling here.

Reply:
I've just noticed: when writing to stdout/stderr fails, all calls to (`e`)`println!` will panic as well, so we'd have to replace them everywhere if we want to not panic in that case. Is that a goal?
I think that not being able to write to standard file descriptors is quite a catastrophic condition, and maybe it is correct to panic here. After all, we can't even print an error message to let the user know something has gone wrong.

Reply:
1. `ColorChoise::Auto` [does not check for tty](https://docs.rs/termcolor/0.3.3/src/termcolor/lib.rs.html#140-144), e.g. `cargo add foo >> add.log` will still use colors and will add some escape sequences nonsense to add.log, see what cargo [does](https://github.com/rust-lang/cargo/blob/2316ca0b7d6b756a4fb85f3a32a7c12bf6e162c9/src/cargo/core/shell.rs#L280).
2. I guess it is fine to panic if (`e`)`println!` panics.